frontend/src/pages/featureStore/screens/entities/EntitiesDetails/EntityDetailsView.tsx

```diff
@@ -0,0 +1,113 @@
+  DescriptionListGroup,
+  DescriptionListTerm,
+  Flex,
+  FlexItem,
+  PageSection,
+  Title,
+} from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { Entity } from '#~/pages/featureStore/types/entities.ts';
+import FeatureStoreTags from '#~/pages/featureStore/components/FeatureStoreTags.tsx';
+import FeatureStoreCodeBlock from '#~/pages/featureStore/components/FeatureStoreCodeBlock.tsx';
+import FeatureStoreTimestamp from '#~/pages/featureStore/components/FeatureStoreTimestamp.tsx';
+
+type DetailsItemProps = {
+  label: string;
+  value: React.ReactNode;
+  testId?: string;
+};
+
+const DetailsItem: React.FC<DetailsItemProps> = ({ label, value, testId }) => (
+  <DescriptionListGroup>
+    <DescriptionListTerm data-testid={testId ? `${testId}-label` : undefined}>
+      {label}
+    </DescriptionListTerm>
+    <DescriptionListDescription data-testid={testId ? `${testId}-value` : undefined}>
+      {value}
+    </DescriptionListDescription>
+  </DescriptionListGroup>
+);
+
+type EntityDetailsViewProps = {
+  entity: Entity;
+};
+
+const EntityDetailsView: React.FC<EntityDetailsViewProps> = ({ entity }) => (
+  <PageSection
+    hasBodyWrapper={false}
+    isFilled
+    padding={{ default: 'noPadding' }}
+    isWidthLimited
+    style={{ maxWidth: '75%' }}
+  >
+    <Flex direction={{ default: 'column' }} gap={{ default: 'gap2xl' }}>
+      <FlexItem>
+        <DescriptionList isCompact isHorizontal>
+          <DetailsItem label="Join key" value={entity.spec.joinKey} testId="entity-join-key" />
+          <DetailsItem
+            label="Value type"
+            value={entity.spec.valueType || '-'}
+            testId="entity-value-type"
+          />
+        </DescriptionList>
+      </FlexItem>
+      <FlexItem>
+        <Title headingLevel="h3" data-testid="entity-data-source" style={{ margin: '1em 0' }}>
+          Data source
+        </Title>
+        {entity.dataSources && entity.dataSources.length > 0 ? (
+          <Table aria-label="Properties table" data-testid="properties-table" variant="compact">
+            <Thead>
+              <Tr>
+                <Th>Source Type</Th>
+                <Th>File Url</Th>
+                <Th>Created date</Th>
+                <Th>Last modified date</Th>
+              </Tr>
+            </Thead>
+            <Tbody>
+              {entity.dataSources.map((dataSource) => (
+                <Tr key={dataSource.name}>
+                  <Td>{dataSource.type.split('_')[0]}</Td>
+                  <Td>{dataSource.fileOptions?.uri || '-'}</Td>
+                  <Td>
+                    <FeatureStoreTimestamp date={dataSource.meta.createdTimestamp} />
+                  </Td>
+                  <Td>
+                    <FeatureStoreTimestamp date={dataSource.meta.lastUpdatedTimestamp} />
+                  </Td>
+                </Tr>
+              ))}
+            </Tbody>
+          </Table>
+        ) : (
```

Comment:
This will be a description list not a table
<img width="1059" height="164" alt="Screenshot 2025-07-31 at 11 52 40 AM" src="https://github.com/user-attachments/assets/93ebc1d3-0fce-438f-a33f-33fba8a59002" />

Reply:
https://www.patternfly.org/components/description-list#horizontal